MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -2672,9 +2672,9 @@ private string GetFixedSizeScalingFilter(Video3DFormat? threedFormat, int reques
                 filters.Add("hwupload");
             }
 
-            // When burning in graphical subtitles using overlay_qsv, upload videostream to the same qsv context.
-            else if (isLinux && hasGraphicalSubs && (isQsvH264Encoder || isQsvHevcEncoder)
-                     && !(isTonemappingSupportedOnQsv && isVppTonemappingSupported))
+            // When burning in graphical subtitles using overlay_qsv or software decoding, upload videostream to the same qsv context.
+            else if ((isLinux && hasGraphicalSubs && (isQsvH264Encoder || isQsvHevcEncoder)
+                     && !(isTonemappingSupportedOnQsv && isVppTonemappingSupported)) || isSwDecoder)
```

Comment:
Seems reasonable to require extra_hw_frames based on this https://trac.ffmpeg.org/ticket/7120 and https://trac.ffmpeg.org/wiki/Hardware/QuickSync.

_But_ this check doesn't seem entirely valid. Should it not only add it if it's Linux _and_ it uses Qsv HW Encoder?

This seems more in line with what I'd expect, but I could be wrong... @nyanmisaka is the authority here.
```suggestion
            else if (isLinux && (hasGraphicalSubs || isSwDecoder) && (isQsvH264Encoder || isQsvHevcEncoder)
                     && !(isTonemappingSupportedOnQsv && isVppTonemappingSupported))
```

Reply:
I just tested it, the same issue occurs on Windows and the same fix applies.

Reply:
so maybe it would be more correct to remove the isLinux check?

```
            else if ((hasGraphicalSubs || isSwDecoder) && (isQsvH264Encoder || isQsvHevcEncoder)
                     && !(isTonemappingSupportedOnQsv && isVppTonemappingSupported))
```

Reply:
I can't test graphical subtitles, so I've changed the condition so that the logic stays unchanged:

`            else if (((isLinux && hasGraphicalSubs && !(isTonemappingSupportedOnQsv && isVppTonemappingSupported))
                || isSwDecoder) && (isQsvH264Encoder || isQsvHevcEncoder))`